### PR TITLE
add classmethods for all vectorizer options

### PR DESCRIPTION
**Pull request by tsmith023**
This PR completes the implementation of the `VectorizerFactory` class by creating the remaining `@classmethod`s for all the available module configs and removing the re-export for the raw Pydantic classes from `weaviate.weaviate_classes

### CODE REVIEW

File: weaviate/weaviate_classes.py [hunk 1]
```diff
@@ -64,29 +51,16 @@
...
-    "Text2VecCohereConfig",
-    "Text2VecGPT4AllConfig",
-    "Text2VecHuggingFaceConfig",
-    "Text2VecHuggingFaceConfigOptions",
-    "Text2VecOpenAIConfig",
-    "Text2VecPalmConfig",
-    "Text2VecTransformersConfig",
     "Tokenization",
     "Vectorizer",
     "VectorizerConfig",
```
Comment (dirkkul) [inline weaviate/weaviate_classes.py:66]: factory is missing here and I think we can remove Vectorizer and VectorizerConfig

Comment (tsmith023) [reply weaviate/weaviate_classes.py:66]: I think `VectorizerConfig` is required for them when using a custom vectorizer, e.g. some standalone Docker container or personal API

Comment (tsmith023) [reply weaviate/weaviate_classes.py:66]: `VectorizerFactory` is already there in un-diffed code on L15 ;)

Comment (tsmith023) [reply weaviate/weaviate_classes.py:64]: Ahhh, you're quite right! The type of `VectorizerConfig.vectorizer` should be `Union[Vectorizer, str]` to allow for users to specify their own named vectorizer as a raw string. WDYT?

Comment (dirkkul) [reply weaviate/weaviate_classes.py:64]: let's remove it for now and leave support for custom modules it for later. We might also need to add a way to set a moduleconfig

**Review by dirkkul — APPROVED** [review]

**Pull request merged**